### PR TITLE
Correct possible misalignment of m_bIsStarted

### PR DESCRIPTION
**Pull request by gordonb3**
- override value of m_bIsStarted in CDomoticzHardware:
  most hardware classes set m_bIsStarted unconditionally to true  where it should  be the return value of StartHardware()
- change Start() and Stop() functions return type to void:
  their return is never used
- add Restart() function
- allow passing thread name to StartHeartbeatThread()

### CODE REVIEW

File: hardware/DomoticzHardware.cpp [hunk 1]
```diff
@@ -26,15 +26,32 @@ bool CDomoticzHardwareBase::CustomCommand(const uint64_t /*idx*/, const std::str
...
+}
+
+void CDomoticzHardwareBase::RestartWithDelay(const long seconds)
+{
+	if (StopHardware())
+	{
+		sleep_seconds(seconds);
+		m_bIsStarted = StartHardware();
+	}
 }
```
Comment (emontnemery) [inline hardware/DomoticzHardware.cpp]: From which context will this function be called?

File: hardware/DomoticzHardware.cpp [hunk 2]
```diff
@@ -65,11 +87,9 @@ void CDomoticzHardwareBase::Do_Heartbeat_Work()
 {
 	int secCounter = 0;
 	int hbCounter = 0;
-	while (!m_stopHeartbeatrequested)
+	while (!IsStopRequested(1000))
```
Comment (gizmocuz) [inline hardware/DomoticzHardware.cpp]: If you use IsStopRequested(), you should use the same value for the sleep_milliseconds, and delete the sleep function

File: hardware/DomoticzHardware.cpp [hunk 3]
```diff
@@ -26,15 +26,32 @@ bool CDomoticzHardwareBase::CustomCommand(const uint64_t /*idx*/, const std::str
 	return false;
 }
 
-bool CDomoticzHardwareBase::Start()
+void CDomoticzHardwareBase::Start()
```
Comment (gizmocuz) [inline hardware/DomoticzHardware.cpp]: please keep the boolean as also with the stop.
If it is not used, ok, but at least we also return true/false.

**Comment by gizmocuz** [comment]
Thanks for the PR.
Could you remove the 'RestartWithDelay' and make it optional for the 'Restart' function ? ( Restart(const int seconds=0) )
(int should be fine)

The heartbeat thread in the basehardware class, can you make it not depending on the default StoppableTask class, but add a StoppableMember, like i did in some hardware classes (1Wire.h/cpp, StoppableTask m_TaskSwitches)

**Comment by gizmocuz** [comment]
Thanks!!

**Pull request merged**